File: classefuncionario.py
import unittest

class TestFuncionario(unittest.TestCase):
    def test_calculo_inss(self):
        funcionario = Funcionario("Maria", 6000)
        self.assertEqual(funcionario.calcular_inss(), 600)

    def test_inss_abaixo_limite(self):
        funcionario = Funcionario("João", 4000)
        self.assertEqual(funcionario.calcular_inss(), 0)

if _name_ == "_main_":
    unittest.main()


class Funcionario:
    def _init_(self, nome, salario):
        self.nome = nome
        self.salario = salario

    def calcular_inss(self):
        if self.salario > 5000:
            return self.salario * 0.10
        return 0

File: classefuncionario.java
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

class FuncionarioTest {
    
    @Test
    void testCalcularInss() {
        Funcionario funcionario = new Funcionario("Maria", 6000);
        assertEquals(600, funcionario.calcularInss());
    }

    @Test
    void testInssAbaixoLimite() {
        Funcionario funcionario = new Funcionario("João", 4000);
        assertEquals(0, funcionario.calcularInss());
    }
}


public class Funcionario {
    private String nome;
    private double salario;

    public Funcionario(String nome, double salario) {
        this.nome = nome;
        this.salario = salario;
    }

    public double calcularInss() {
        if (this.salario > 5000) {
            return this.salario * 0.10;
        }
        return 0;
    }

    public String getNome() {
        return nome;
    }

    public double getSalario() {
        return salario;
    }
}